svg/include/svg/svg.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <sstream>
+
+#include "element.hpp"
+
+namespace svg::image {
+
+    class SVGDimension {
+    public:
+        SVGDimension(double width, double height);
+
+        double get_width() const;
+        double get_height() const;
+
+    private: 
+        double width;
+        double height;
+    };
+
+    class SVG {
+    public:
+        virtual ~SVG();
+
+        SVG(const SVGDimension& dimension, const std::string name);
+        
+        void const add_element(const geometry::Element& element);
```

Comment:
When possible, it's always preferable to have immutable class. It's much easier to reason with and you can avoid bugs using the type system. So you could introduce a `SvgStub` class who will act as incomplete svg (a bit like a builder pattern) or you could pass to a static class a list of elements.